tests/warnings_util_test.py

```diff
@@ -0,0 +1,81 @@
+# Copyright 2024 The JAX Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import warnings
+
+from absl.testing import absltest
+
+from jax._src import config
+from jax._src import test_warning_util
+from jax._src import test_util as jtu
+
+
+config.parse_flags_with_absl()
+
+class WarningsUtilTest(jtu.JaxTestCase):
+
+  def test_warning_raises(self):
+    with self.assertRaises(UserWarning, msg="hello"):
+      warnings.warn("hello", category=UserWarning)
```

Comment:
Could we also add a similar test `DeprecationWarning`? It has some different handing than `UserWarning` within Python's warning system, and it's important that we still error when we see it.